public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -107,11 +107,13 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   useEffect(() => {
     const currentCondition = getValues('condition');
 
-    if (!currentCondition || RuleFormType.cloudRecording) {
+    if (!currentCondition || !queryPreviewData[currentCondition]) {
```

Comment:
`RuleFormType.cloudRecording` always evaluated to `true` because it's just a string value :)